WordPress/src/test/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostDetailViewModelTest.kt

```diff
@@ -772,6 +791,80 @@ class ReaderPostDetailViewModelTest {
         )
     }
 
+    @Test
+    fun `likes for post are refreshed on request`() = test {
+        val likesState = getGetLikesState(TEST_CONFIG_1) as LikesData
+
+        getLikesState.value = likesState
+        init()
+        viewModel.onRefreshLikersData(viewModel.post!!)
+        verify(getLikesHandler, times(1)).handleGetLikesForPost(anyOrNull(), anyBoolean(), anyInt(), anyInt())
+    }
+
+    @Test
+    fun `ui state show likers faces when data available`() {
+        val likesState = getGetLikesState(TEST_CONFIG_1) as LikesData
+        val likers = MutableList(5) { mock<Liker>() }
+
+        getLikesState.value = likesState
+        whenever(engagementUtils.likesToEngagedPeople(anyList(), eq(null), eq(null))).thenReturn(likers)
+        val post = mock<ReaderPost>()
+        whenever(post.isWP).thenReturn(true)
+        viewModel.post = post
+        val likeObserver = init().likesUiState
+
+        getLikesState.value = likesState
+
+        assertThat(likeObserver).isNotEmpty
+        with(likeObserver.first()) {
+            assertThat(showLikeFacesTrain).isTrue
+            assertThat(showLoading).isFalse
+            assertThat(engageItemsList).isEqualTo(likers)
+            assertThat(numLikes).isEqualTo(likesState.expectedNumLikes)
+            assertThat(showEmptyState).isFalse
+            assertThat(emptyStateTitle).isNull()
+            assertThat(likersFacesText is UiStringText).isTrue
+        }
+    }
+
+    @Test
+    fun `ui state shows empty state on failure and no cached data`() {
+        val likesState = getGetLikesState(TEST_CONFIG_5) as Failure
+        val likers = listOf<Liker>()
+
+        getLikesState.value = likesState
+        whenever(engagementUtils.likesToEngagedPeople(anyList(), eq(null), eq(null))).thenReturn(likers)
+        val post = mock<ReaderPost>()
+        whenever(post.isWP).thenReturn(true)
+        viewModel.post = post
+        val likeObserver = init().likesUiState
+
+        getLikesState.value = likesState
+
+        assertThat(likeObserver).isNotEmpty
+        with(likeObserver.first()) {
+            assertThat(showLikeFacesTrain).isTrue
```

Comment:
Not sure, but shouldn't `showLikeFacesTrain` be false on failure and no cache?

Reply:
So, I think we should rename the `showLikeFacesTrain` to something like `showLikeFacesTrainContainer`; it is actually the root container so it should be visible to show the empty state, but agree the naming is not good 👍 . I'm going to open a small separate PR for that if you agree 🙇

Reply:
Sounds good 👍